File: academic-system/src/main/java/org/example/model/Turma.java
package org.example.model;
import java.util.List;
import java.util.ArrayList;

public class Turma {
	
	private String codigo;
	private String nome;
	private ArrayList<Avaliacao> avaliacoes;

	
	

    public Turma(String codigo, String nome) {
    	this.codigo = codigo;
    	this.nome = nome;
    	this.avaliacoes = new ArrayList<>();
    	
    }

    
    //AC6: Dados inválidos de avaliação,
    //quando uma tentativa de registro de avaliação é feita,
    //então o sistema deve rejeitar a operação lançando uma AcademicSystemException.
    public void adicionarAvaliacao(Avaliacao avaliacao) { 
    	if(avaliacao == null) {
    		throw new IllegalArgumentException("A avaliação não pode ser nula");
    	}
    	this.avaliacoes.add(avaliacao);
    	
    }

    public String getCodigo() {return codigo;}
    public String getnome() {return nome;}
    
    public List<Avaliacao> getAvaliacoes() { 
        return new ArrayList<>(avaliacoes); 
    }
}

File: academic-system-csharp/SistemaAcademico/src/SistemaAcademico/Models/Turma.cs
using System.ComponentModel.DataAnnotations;
using SistemaAcademico.Exceptions;

namespace SistemaAcademico.Models;

// Equivalente à classe Java: Turma
// Lombok @Getter → propriedades C# com get; privado
// Lombok @EqualsAndHashCode(onlyExplicitlyIncluded = true) com @EqualsAndHashCode.Include em codigo
//   → Equals e GetHashCode sobrescritos baseados apenas em Codigo
public class Turma
{
    [Required(ErrorMessage = "O código da turma não pode ser vazio.")]
    [MinLength(1, ErrorMessage = "O código da turma não pode ser vazio.")]
    public string Codigo { get; private set; }

    [Required(ErrorMessage = "O título da turma não pode ser vazio.")]
    [MinLength(1, ErrorMessage = "O título da turma não pode ser vazio.")]
    public string Titulo { get; private set; }

    private readonly List<Avaliacao> avaliacoes;

    public Turma(string codigo, string titulo)
    {
        Codigo = codigo;
        Titulo = titulo;
        avaliacoes = new List<Avaliacao>();
    }

    // AC6: Dados inválidos de avaliação – rejeitar nulo com IllegalArgumentException
    public void AdicionarAvaliacao(Avaliacao avaliacao)
    {
        if (avaliacao == null)
        {
            throw new ArgumentException("A avaliação não pode ser nula");
        }
        avaliacoes.Add(avaliacao);
    }

    // Retorna cópia defensiva da lista (equivalente ao new ArrayList<>(avaliacoes) Java)
    public List<Avaliacao> GetAvaliacoes()
    {
        return new List<Avaliacao>(avaliacoes);
    }

    // Equivalente a Lombok @EqualsAndHashCode(onlyExplicitlyIncluded = true)
    // Igualdade baseada apenas no campo Codigo
    public override bool Equals(object? obj)
    {
        if (obj is Turma other)
        {
            return string.Equals(Codigo, other.Codigo, StringComparison.Ordinal);
        }
        return false;
    }

    public override int GetHashCode()
    {
        return Codigo?.GetHashCode() ?? 0;
    }
}
